plugins/inputs/mesos/iam.go

```diff
@@ -0,0 +1,75 @@
+package mesos
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/dcos/dcos-go/dcos/http/transport"
+)
+
+type DCOSConfig struct {
+	CACertificatePath string `toml:"ca_certificate_path"`
+	IAMConfigPath     string `toml:"iam_config_path"`
+}
+
+// transport returns a transport implementing http.RoundTripper
+func (c *DCOSConfig) transport() (http.RoundTripper, error) {
+	tr, err := getTransport(c.CACertificatePath)
+	if err != nil {
+		return nil, err
```

Comment:
I think we should add context to this error eg. `return nil, fmt.Errorf("could not create transport: %s", err)`

Reply:
I added it to the higher level call [here](https://github.com/dcos/telegraf/pull/23/files#diff-214de6d5d68d1fa8e34875623475b29aR313)